presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanChecker.java

```diff
@@ -17,27 +17,41 @@
 import com.facebook.presto.metadata.Metadata;
 import com.facebook.presto.spi.WarningCollector;
 import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.SimplePlanFragment;
 import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.sanity.plancheckerprovidermanagers.PlanCheckerProviderManager;
 import com.google.common.collect.ImmutableListMultimap;
 import com.google.common.collect.Multimap;
 
 import javax.inject.Inject;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 /**
  * Perform checks on the plan that may generate warnings or errors.
  */
 public final class PlanChecker
 {
-    private final Multimap<Stage, Checker> checkers;
+    private final PlanCheckerProviderManager providerMgr;
+    private Multimap<Stage, Checker> checkers;
+    private boolean providedInitialized;
 
     @Inject
-    public PlanChecker(FeaturesConfig featuresConfig)
+    public PlanChecker(FeaturesConfig featuresConfig, PlanCheckerProviderManager providerManager)
     {
-        this(featuresConfig, false);
+        this(featuresConfig, false, providerManager);
     }
 
     public PlanChecker(FeaturesConfig featuresConfig, boolean forceSingleNode)
     {
+        this(featuresConfig, forceSingleNode, null);
```

Comment:
This should never be allowed to be null. We avoid nulls whenever possible in Presto. Instead have it be empty.